src/js/utils/axis-chart-utils.js

```diff
@@ -33,6 +33,19 @@ export function dataPrep(data, type) {
 			}
 		}
 
+		//XSS
+		try{
+			for(var i=0;i<datasetLength;i++){
+				var temp = data.datasets[i]['values'];
+				for(var j=0;j<temp.length;j++){
+					if(typeof(temp[j]) == "string"){
+						data.datasets[i]['values'][j] = String(temp[j]).replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#039;");
+					}
+				}
```

Comment:
The loop is really inefficient, let's just check if the item is a valid number or not, there's a utility for that

Reply:
Yes it check whether its number or not, But actual problem is not does not work properly. even script are inserted into charts and that leads to XSS

Reply:
> even Strings are inserted into charts

The values should always be numbers, strings shouldn't be allowed. For labels a fix for this was already made [here](https://github.com/frappe/charts/pull/315)